### ISSUE TITLE
Failure to init/createWindow after a previous session

### ISSUE THREAD

**Issue by plredmond** [bsl/GLFW-b#71]
### Summary
If GLFW-b is stood up and torn down on the same main-thread multiple times it may fail to create windows. The issue seems to be connected to event handling.

### Repro
Write a program which initializes and deinitializes GLFW-b multiple times on the same thread. For extra reproducability, try posting and waiting on events a lot.

1. [Add this test to your local branch](https://github.com/bsl/GLFW-b/compare/master...plredmond:repro#diff-522957e9c653d0c47d6ff5944bc25d5a)
2. Build and run the test with `stack`
```
$ stack test GLFW-b:two
GLFW-b-1.4.8.1: test (suite: two)

Testing multiple initializations of GLFW-b
Set up and tore down GLFW-b once
two: GlfwCreateWindowException

Test suite failure for package GLFW-b-1.4.8.1
    two:  exited with: ExitFailure 1
Logs printed to console
```

#### Expected behavior
GLFW initializes, creates a window, destroys the window, and deinitializes.
GLFW again initializes, creates a window, destroys the window, and deinitializes.

#### Actual behavior
GLFW fails to return a result from `createWindow` on the second call.

### Details
I'm using stack, and the stack.yaml on master hasn't been updated in a year.
My test machine is running OSX on a mid 2015 MacbookPro with two graphics cards.

**Comment by plredmond** [comment]
I think this is related to OSX "automatic graphics switching" feature.

**Comment by Mokosha** [comment]
This bug is now over a year old. It's likely that this was fixed upstream in GLFW in 3.2. For that reason I'm going to close it here.

If it's still a problem, please verify that it is due to the way that the GLFW-b bindings are implemented and not simply a GLFW bug or a FFI bug (by trying to use the raw bindings in [bindings-GLFW](https://hackage.haskell.org/package/bindings-GLFW)). If it is indeed a GLFW-b bug, then feel free to reopen this issue.
